ledger-core/virtual/integration/constructor_test.go

```diff
@@ -548,3 +549,111 @@ func TestVirtual_CallConstructorFromConstructor(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
+	t.Log("C5085")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+		callFlags = payload.BuildCallFlags(isolation.Interference, isolation.State)
+
+		class     = gen.UniqueGlobalRef()
+		outgoing  = server.RandomLocalWithPulse()
+		objectRef = reference.NewSelf(outgoing)
+
+		constructorPulse = server.GetPulse().PulseNumber
+	)
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      callFlags,
+		CallAsOf:       constructorPulse,
+		Callee:         class,
+		CallSiteMethod: "test",
+		CallOutgoing:   outgoing,
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+		assert.Equal(t, outgoing, report.Object.GetLocal())
+		assert.Equal(t, constructorPulse, report.AsOf)
+		assert.Equal(t, payload.Empty, report.Status)
+		assert.Equal(t, int32(1), report.OrderedPendingCount)
+		assert.Equal(t, int32(0), report.UnorderedPendingCount)
+		return true
+	})
+	typedChecker.VDelegatedCallRequest.SetResend(true)
```

Comment:
whats for?

Reply:
token logic is outside of the test case. And without mocking token we cannot make sure that token is send (because it won't be sent due to same node execution)

Reply:
done